src/shared/components/oncoprint/SortUtils.ts

```diff
@@ -302,6 +303,42 @@ export function getClinicalTrackSortComparator(track: ClinicalTrackSpec) {
     };
 }
 
+export function getClinicalTrackInitialSortDirection(
+    track: ClinicalTrackSpec
+): 0 | 1 | -1 {
+    let initialSortOrder: 0 | 1 | -1 = 0;
+
+    const defaultTracks = ServerConfigHelpers.parseDefaultOncoprintClinicalTracks(
+        getServerConfig().oncoprint_clinical_tracks_show_by_default!
+    );
+
+    defaultTracks.forEach(t => {
+        if (t.stableId === track.attributeId) {
+            if (t.sortOrder === 'ASC') {
+                initialSortOrder = 1;
+            } else if (t.sortOrder === 'DESC') {
+                initialSortOrder = -1;
+            }
+        }
+    });
+    return initialSortOrder;
+}
+
+export function getClinicalTrackInitialShowGapsConfig(
+    track: ClinicalTrackSpec
+): boolean {
+    let gapOn = false;
+    const defaultTracks = ServerConfigHelpers.parseDefaultOncoprintClinicalTracks(
+        getServerConfig().oncoprint_clinical_tracks_show_by_default!
+    );
+    defaultTracks.forEach(t => {
```

Comment:
again, should just use find.    performance insignificant but for code clarity

Reply:
see https://github.com/cBioPortal/cbioportal-frontend/pull/4152#discussion_r842880223